customTests/custom-test-setup.sh

```diff
@@ -73,35 +73,43 @@ function install_stream(){
 	cp ${SRC_DIR}/customTests/stream/*  $STREAM_DIR
 	pushd $STREAM_DIR
 
-	# Stream
-	if command -v /opt/AMD/aocc-compiler-4.0.0/bin/clang &> /dev/null || command -v clang &> /dev/null; then
-		echo -e "clang compiler found Building Stream"
+	# default path
+	CLANG=$( find /opt/azurehpc/spack/ -name "clang" | grep 'bin' )
+
+	if ! command -v "$CLANG" &> /dev/null; then
+		# for older images
+		CLANG=$( find /opt/AMD -name "clang" | grep 'bin' )
+	fi
 
+	if command -v clang &> /dev/null &&  ! command -v "$CLANG" &> /dev/null ; then
```

Comment:
Isn't the second clause here the same as the previous IF?

Reply:
Kinda.
1. first we set  the $CLANG to the Spack path (line 77)
2. First if condition: check if the $CLANG command exists, if not try to set $CLANG to old OS image path
3. Second if condition (line 84):  In the event that $CLANG is still not a command then we check if it is part of the path. If "clang" is part of path set $CLANG to the "clang in path"
4. last condition (line 88): now that we have tried ever combination of $CLANG lets check to see for the last time if it is command.

Let me see if I can refactor this to make it better